SparkPost/2web2ui
# SA-1481: Fixed bug of report initializing twice

### What Changed
 - Added check for fully initializing params of report options before making request
   - Change comes from the translation from Class to functional with lifecycle methods to hooks

### How To Test
 - Load Report builder with options different from default
 - Refresh page
 - Check for network requests. There should only be one initial time series instead of 2

### To Do
- [ ] See if there's a better pattern for initialization rather than a flag
   - Could go with the option of making it a string enum.
